alloy/src/main/java/com/liferay/faces/alloy/component/head/internal/HeadRenderer.java

```diff
@@ -43,31 +50,49 @@
 		}
 	)
 //J+
-public class HeadRenderer extends HeadRendererBase {
+@ListenerFor(systemEventClass = PostAddToViewEvent.class, sourceClass = Head.class)
+public class HeadRenderer extends HeadRendererBase implements ComponentSystemEventListener {
 
 	// Private Constants
 	private static final boolean LIFERAY_PORTAL_DETECTED = ProductFactory.getProduct(Product.Name.LIFERAY_PORTAL)
 		.isDetected();
 
 	@Override
-	public void encodeChildren(FacesContext facesContext, UIComponent uiComponent) throws IOException {
+	public void processEvent(ComponentSystemEvent componentSystemEvent) throws AbortProcessingException {
 
 		// If Liferay Portal is not detected, then encode a meta tag as a child of the head that will cause bootstrap
 		// to behave responsively.
 		if (!LIFERAY_PORTAL_DETECTED) {
 
+			FacesContext facesContext = FacesContext.getCurrentInstance();
+			UIViewRoot uiViewRoot = facesContext.getViewRoot();
+			ResponsiveMetadata responsiveMetadata = new ResponsiveMetadata();
+			Map<String, Object> attributes = responsiveMetadata.getAttributes();
+			attributes.put("name", ResponsiveMetadata.COMPONENT_FAMILY);
+			uiViewRoot.addComponentResource(facesContext, responsiveMetadata);
+		}
+	}
+
+	public static class ResponsiveMetadata extends UIComponentBase {
```

Comment:
@stiemannkj1 Please send a follow-up PR with `saveState` and `restoreState` methods. As I recall we have had problems in the past dynamically adding components to the component tree that did not know how to save/restore themselves.

Reply:
@stiemannkj1 Also, please send a follow-up PR for the 2.x branch since this PR would not backport without conflicts.

Reply:
@ngriffin7a, we only really need to worry about save/restore state when there is actual state in the component. In the case of `ResponsiveMetadata` the component has no state.

For reference take a look at the fix for `InlineScript`: https://github.com/liferay/liferay-faces-bridge-impl/commit/c01aa28c49b85fc8f9badd89695a706e2b5e247e#diff-7914f0ffdd891a368807629fb4138a37. We only made that change because of the private member variable: `script`.